### DISCUSSION TITLE
Advice on using pg_dump and pg_restore for OSM dataset

### DISCUSSION THREAD

**Discussion by vidushirai** [osm-search/Nominatim#3076] [Q&A]
👋 hello! 
We self host nominatim on ec2 instances and these instances are launched using an AMI that was built a year ago. I'm currently working on creating a new AMI on a different architecture. However, I don't want to import the latest OSM dataset - instead I want the revgeo results from the new AMI to 100% match the revgeo results from the original AMI. 
In order to do this, I ran the pg_dump command on an instance running with the original AMI. 

```
sudo docker exec -it postgres_docker_container pg_dump -U username nominatim_database | gzip > /backup/nominatim.gz
```

I then wrote this dump file to S3. In the new nominatim AMI, I skip the import commands and instead ran 

```
aws s3 cp s3://bucket/nominatim.gz - | gunzip -c | sudo docker exec -i postgres_docker_container sh -c 'psql -U username nominatim'
```

This worked to the extent where the nominatim database was created and its size matched the size of the database in instances with the original AMI. However, when I do a comparison of revgeo results for a sample set of coordinates, I end up with ~0.03% of the revgeo results not matching which is confusing to me. 

Wondering if anyone else has tried this approach - is pg_dump and restore not an accurate way to replicate the information in the database? 

Notes: 
- in the original AMI we also download https://www.nominatim.org/data/gb_postcode_data.sql.gz and https://www.nominatim.org/data/us_postcode_data.sql.gz files 
- we also import additional tiger data: 
```
 wget https://nominatim.org/data/tiger2020-nominatim-preprocessed.tar.gz
 nominatim add-data --tiger-data tiger2020-nominatim-preprocessed.tar.gz --threads "$(nproc)"
 nominatim refresh --functions --threads "$(nproc)"
```

I added these lines to my new AMI and the revgeo results mismatch somehow grew to 33%. Also, from reading the docs ([link](https://nominatim.org/release-docs/3.7.0/admin/Import/#installing-tiger-housenumber-data-for-the-us), [link](https://nominatim.org/release-docs/3.7.0/admin/Import/#great-britain-usa-postcodes)) it seems like the information in these files would get added to the database regardless. 

Would appreciate any advice / help!

**Comment by lonvia** [comment]
pg_dump/pg_restore should work just fine. You do need to run`nominatim refresh --functions` after the restore. The 0.03% non-matching results may just be cases where there are two equal answers and the database returns one randomly. For reverse geocoding this can happen if two objects are exactly at the same distance from the requested coordinate. I wouldn't worry about it.

The Tiger and postcode data will be part of the dump, so no need to reimport.
